jetty-websocket/websocket-jetty-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java

```diff
@@ -83,6 +83,26 @@ public WebSocketClient()
      */
     public WebSocketClient(HttpClient httpClient)
     {
+        this (httpClient,
+            new WebSocketComponents(
+                null,
+                null,
+                httpClient != null ? httpClient.getByteBufferPool() : null,
+                null,
+                null,
+                httpClient != null ? httpClient.getExecutor() : null)
+        );
+    }
+
+    /**
+     * Instantiate a WebSocketClient using HttpClient for defaults
+     *
+     * @param httpClient the HttpClient to base internal defaults off of
```

Comment:
The javadoc for this is wrong, because in this constructor we already have a `WebSocketComponents`. 

Also for the other constructor you should say how we actually use the `HttpClient` for upgrade requests (its not just to base internal defaults).